bin/analysis.cpp

```diff
@@ -237,6 +236,20 @@ int main(int ac, char* av[])
 		return 1;
 	}
 
+
+	std::cout << "[INFO] Selected modes: \n";
+	unsigned int msize = modes.size();
+	modes.erase( std::remove( modes.begin(), modes.end(), "signal" ), modes.end() );
+	if(modes.size() != msize){
```

Comment:
Here you are adding all the signal samples, why this cannot be done at line 118 if mode==signal?